sdk/core/azure-core/inc/azure/core/http/http.hpp

```diff
@@ -662,7 +665,7 @@ namespace Azure { namespace Core { namespace Http {
      *
      * @throw if \p name contains invalid characters.
      */
-    void AddHeader(std::string const& name, std::string const& value);
+    void SetHeader(std::string const& name, std::string const& value);
```

Comment:
Update the docs for this and other `SetHeader` overloads as well, similar to `Request::SetHeader()` above, to remark about existing/new header names and what happens in those cases.

Also update to use `Set` instead of `Add` in all the doc comments for consistency.